convert2rhel/main.py

```diff
@@ -168,26 +165,33 @@ def main_locked():
 
         rollback_changes()
 
-        report.summary(
+        report.pre_conversion_report(
             pre_conversion_results,
             include_all_reports=True,
             disable_colors=logger_module.should_disable_color_output(),
         )
         return 0
-
     except exceptions.CriticalError as err:
         loggerinst.critical_no_exit(err.diagnosis)
-        return _handle_main_exceptions(process_phase, pre_conversion_results)
-
+        results = _pick_conversion_results(process_phase, pre_conversion_results, post_conversion_results)
+        return _handle_main_exceptions(process_phase, results)
     except (Exception, SystemExit, KeyboardInterrupt) as err:
-        return _handle_main_exceptions(process_phase, pre_conversion_results)
-
+        results = _pick_conversion_results(process_phase, pre_conversion_results, post_conversion_results)
+        return _handle_main_exceptions(process_phase, results)
```

Comment:
Would `post_conversion_result` be populated even if we're not in post-ponr phase?

Given it's only used here though we could allow some duplication if we are going to refactor it in the future, instead of a function that is difficult to define